src/components/dialogs/illustrations/generic-Illustration-network-modification.tsx

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useIntl } from 'react-intl';
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import Paper from '@mui/material/Paper';
+import { LIGHT_THEME } from '@gridsuite/commons-ui';
+import { AccordionIllustration } from '../network-modifications/widget/AccordionIllustration';
+import { PARAM_THEME } from 'utils/config-params';
+import { ReduxState } from 'redux/reducer.type';
+
+interface ReplacedText {
+    eltId: string;
+    tradId: string;
+}
+
+interface GenericIllustrationNetworkModificationProps {
+    svgComponent: React.ComponentType<any>;
+    replacedTexts: ReplacedText[];
+    backgroundElementId: string;
+}
+/**
+ * This component displays a hide able explanation illustration
+ */
+const GenericIllustrationNetworkModification: React.FC<
+    GenericIllustrationNetworkModificationProps
+> = ({ svgComponent: SvgComponent, replacedTexts, backgroundElementId }) => {
+    const [showDiagram, setShowDiagram] = useState(true);
+    const intl = useIntl();
+    const theme = useSelector((state: ReduxState) => state[PARAM_THEME]);
+
+    useEffect(() => {
+        // dynamically modify the text inside the svg :
+        replacedTexts.forEach((replacedText) => {
+            const elt = document.getElementById(replacedText.eltId);
+            if (!!elt) {
+                elt.textContent = intl.formatMessage({
+                    id: replacedText.tradId,
+                });
+            }
+        });
+    }, [intl, replacedTexts]);
+
+    useEffect(() => {
+        /**
+         * updates opacity according to the theme by updating the 'style' string from the svg element
+         * @param propStr : String preceding and describing the property whose opacity must be changed
+         * @param svgElt : element fetched from the svg that should be updated
+         */
+        function updateOpacity(propStr: string, svgElt: HTMLElement | null) {
+            if (!!svgElt) {
+                const eltCssText = svgElt.style.cssText;
+                const indexOpacity = eltCssText.indexOf(propStr);
+                if (indexOpacity !== -1) {
+                    svgElt.style.cssText =
+                        eltCssText.substring(0, indexOpacity + propStr.length) +
+                        (theme === LIGHT_THEME ? '0' : '1') +
+                        eltCssText.substring(indexOpacity + propStr.length + 1);
+                } else {
+                    svgElt.style.cssText =
+                        'fill-opacity: ' + (theme === LIGHT_THEME ? '0' : '1');
```

Comment:
Here 'fill-opacity: ' should be replaced by 'propStr'.

Reply:
done